src/main/java/ch/njol/skript/expressions/ExprNamed.java

```diff
@@ -59,53 +60,93 @@ public class ExprNamed extends PropertyExpression<Object, Object> {
 	@SuppressWarnings("null")
 	private Expression<String> name;
 	
-	@SuppressWarnings({"unchecked", "null"})
 	@Override
-	public boolean init(final Expression<?>[] exprs, final int matchedPattern, final Kleenean isDelayed, final ParseResult parseResult) {
-		setExpr(exprs[0]);
+	@SuppressWarnings("unchecked")
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
+		if (exprs[0].getReturnType().equals(ItemStack.class)) {
+			setExpr(exprs[0].getConvertedExpression(ItemType.class));
+		} else {
+			setExpr(exprs[0]);
+		}
 		name = (Expression<String>) exprs[1];
+		check_type_okay:
+		if (getExpr() instanceof Literal) {
+			Literal<?> literal = (Literal<?>) getExpr();
+			Object object = literal.getSingle();
+			if (!(object instanceof InventoryType)) break check_type_okay;
+			if (!isCreatable((InventoryType) object)) {
```

Comment:
```suggestion
		if (getExpr() instanceof Literal) {
			Literal<?> literal = (Literal<?>) getExpr();
			Object object = literal.getSingle();
			if (object instanceof InventoryType && !isCreatable((InventoryType) object)) {
```
I think this would be preferred